yaksh/views.py

```diff
@@ -614,10 +614,43 @@ def add_course(request):
 
 
 @login_required
+def edit_course(request, course_id=None):
+    user = request.user
+    ci = RequestContext(request)
+    if course_id:
+        course = Course.objects.get(id=course_id)
+    else:
+        course = None
+
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page')
+    if request.method == 'POST':
+        form = CourseForm(request.POST, instance=course)
+        if form.is_valid():
+            new_course = form.save(commit=False)
+            new_course.creator = user
+            new_course.save()
+            return my_redirect('/exam/manage/')
+        else:
+            return my_render_to_response('yaksh/add_course.html',
+                                         {'form': form},
+                                         context_instance=ci)
+    else:
+        form = CourseForm(instance=course)
+        return my_render_to_response('yaksh/add_course.html', {'form': form},
+                                     context_instance=ci)
+
+
+@login_required
 def enroll_request(request, course_id):
     user = request.user
     ci = RequestContext(request)
     course = get_object_or_404(Course, pk=course_id)
+    if not course.is_active_enrollment:
+        msg = 'Enrollment for this course has been closed, please contact your '\
+            'instructor/administrator.'
+        return complete(request, msg, attempt_num, questionpaper_id=None)
```

Comment:
- attempt_num is fixed.
- The ```complete()``` displays the message as an error, the template that is displayed has a button that redirects to dashboard